### PR TITLE
Create post

### PR DESCRIPTION
**Pull request by yspolatt**
I implemented create post endpoint. I**t requires detailed reviewing since I just recently started coding backend. I included tests as well.**

### CODE REVIEW

File: backend/api/views.py [hunk 1]
```diff
@@ -199,4 +199,108 @@ def reset_password(request):
...
         return JsonResponse({'success':'Password updated'})
     else: 
-        return JsonResponse({'error':'No user found'}, status=404)
\ No newline at end of file
+        return JsonResponse({'error':'No user found'}, status=404)
+
+
+@csrf_exempt
+@require_http_methods(["POST"])
+def create_post(request):
```
[the remark this answers is not in the document]
Comment (eyusufatik) [reply backend/api/views.py:207]: but first it should check if the link provided exists in the database.

if exists, link is already validated with type etc. so we can just append the post

File: backend/api/views.py [hunk 2]
```diff
@@ -199,4 +199,108 @@ def reset_password(request):
...
+        for post in posts:
+            posts_data.append({
+                'id': post.id,
+                'content': post.content,
+                'images': post.images,
+                'links': post.links,
+                'timestamp': post.timestamp,
+                'tags': list(post.tags.values_list('name', flat=True))
+            })
+
```
Comment (eyusufatik) [inline backend/api/views.py:268]: we need paging on this endpoint,  cant return all posts from db in a single req

File: backend/music_app/urls.py [hunk 3]
```diff
@@ -28,4 +28,8 @@
     path('api/logout/', views.logout, name='logout'),
     path('api/follow/<int:user_id>/', views.follow_user, name='follow_user'),
     path('api/unfollow/<int:user_id>/', views.unfollow_user, name='unfollow_user'),
+    path('api/create-post/', views.create_post, name='create_post'),
+    path('api/get-posts/', views.get_posts, name='get_posts'),
+    path('api/get-post/<int:post_id>/', views.get_post, name='get_post'),
```
Comment (eyusufatik) [inline backend/music_app/urls.py:33]: I'd do 

POST posts
GET posts -> a single endpoint which can be used like:

/posts?start=<int>&size=<int>

or

/posts?id=<int>

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: backend/api/models.py [hunk 4]
```diff
@@ -47,4 +47,34 @@ def get_followers(self):
...
+    
+
+class Tag(models.Model):
+    name = models.CharField(max_length=100, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+    def __str__(self):
```
Comment (burkaykinik) [inline backend/api/models.py:79]: there should be at most one __str__ function

**Review by burkaykinik — CHANGES_REQUESTED** [review]
My only review in addition to Esad's comments is that only one of the __str__ functions should be keps in the Tag model.

File: backend/api/models.py [hunk 5]
```diff
@@ -47,4 +47,34 @@ def get_followers(self):
...
+    created_at = models.DateTimeField(auto_now_add=True)
+
+class Post(models.Model):
+    id = models.AutoField(primary_key=True)
+    content = models.TextField()
+    images = ArrayField(models.CharField(max_length=200), blank=True, default=list)
+    links = ArrayField(models.CharField(max_length=200), blank=True, default=list)
+    timestamp = models.DateTimeField(default=timezone.now)
+    created_at = models.DateTimeField(auto_now_add=True)
+    likes = models.ManyToManyField(User, related_name='liked_posts', blank=True)
```
[the remark this answers is not in the document]
Comment (eyusufatik) [reply backend/api/models.py:59]: this should stay like this. my bad. how else are we going to know a user already liked the post :D

File: backend/api/views.py [hunk 6]
```diff
@@ -199,4 +199,108 @@ def reset_password(request):
         
         return JsonResponse({'success':'Password updated'})
     else: 
-        return JsonResponse({'error':'No user found'}, status=404)
\ No newline at end of file
+        return JsonResponse({'error':'No user found'}, status=404)
+
+
+@csrf_exempt
```
Comment (eyusufatik) [inline backend/api/views.py:205]: one more thing

only logged in users can call create post

File: backend/api/__pycache__/models.cpython-312.pyc
[the remark this answers is not in the document]
Comment (yspolatt) [reply backend/api/__pycache__/models.cpython-312.pyc:1]: Apparently something went wrong with .gitignore. I will pay more attention next time.

**Comment by yspolatt** [comment]
I committed the changes you pointed to the branch.

**Pull request merged**